var/spack/repos/builtin/packages/octopus/package.py

```diff
@@ -57,8 +59,15 @@ def install(self, spec, prefix):
             'CC=%s' % spec['mpi'].mpicc,
             'FC=%s' % spec['mpi'].mpifc,
             '--enable-mpi',
-            '--with-fftw-prefix==%s' % spec['fftw'].prefix,
         ])
+        if '^fftw' in spec:
+            args.extend([
+                '--with-fftw-prefix=%s' % spec['fftw'].prefix,
+            ])
+        else:
+            args.extend([
+                'FCFLAGS_FFTW=-I%s' % spec['mkl'].prefix.include.fftw
```

Comment:
I don't see a dependency on `mkl`

Reply:
It's a bit convoluted and I should have at least added a comment here.

If FFTW is not part of the dependency tree, then it ought to be MKL because it is currently the only other package that provides `fftw-api` (for which there is a dependency, https://github.com/spack/spack/pull/17782/files#diff-e431dc8274c23dce4f013cf46cd209e3R38).

I am not exactly sure how to make this fool-proof. I could do something like:
```python
elif '^mkl' in spec:
    args.extend([
           FCFLAGS_FFTW=-I%s' % spec['mkl'].prefix.include.fftw
    ])
else:
    # die with an error message explaining that the package needs to be updated
```

What do you think?

Reply:
Ah, I see. Could you override the `headers` attribute of `fftw` and `intel-mkl` so that you could use `spec['fftw-api'].headers.include_flags`?

Reply:
I only needs to define `FCFLAGS_FFTW` when `fftw-api` is provided by MKL, otherwise using `--with-fftw-prefix` is simpler since it also handles the lib dir.

But anyway I am not exactly sure how I could override the `headers` attribute of `intel-mkl` to have the proper behavior when using `spec['fftw-api'].headers.include_flags`. In fact I am not even sure I am understanding correctly what you mean here....

My knowledge of Spack is pretty basic but as far as I know, `fftw-api` is a virtual package that acts as a proxy for an actual package. The actual implementation of the `headers` property of the `intel-mkl` package would have to be specialized depending on whether its is being targeted directly for the package or through a virtual package. I don't think it is currently possible to retrieve this information.

Unless you think `include/fftw` should always be included in the `headers` list provided by the `mkl` package?

Reply:
@adamjstewart: I have repushed a more foolproof version with comments.

I will be happy to change it if you have a better idea on how to handle this properly.